src/main/java/com/commercetools/sync/products/utils/VariantReferenceReplacementUtils.java

```diff
@@ -147,10 +145,25 @@
     }
 
     private static Optional<Set<Reference<Product>>> getProductReferenceSet(@Nonnull final Attribute attribute) {
-        try {
-            return Optional.of(attribute.getValue(AttributeAccess.ofProductReferenceSet()));
-        } catch (final JsonException exception) {
-            return Optional.empty();
+        return isProductReferenceSet(attribute)
+            ? Optional.of(attribute.getValue(AttributeAccess.ofProductReferenceSet())) : Optional.empty();
+    }
+
+    static boolean isProductReference(@Nonnull final Attribute attribute) {
+        final JsonNode valueAsJsonNode = attribute.getValueAsJsonNode();
+        return !(valueAsJsonNode instanceof ArrayNode) && isValueAProductReference(valueAsJsonNode);
+    }
+
+    static boolean isProductReferenceSet(@Nonnull final Attribute attribute) {
+        final JsonNode valueAsJsonNode = attribute.getValueAsJsonNode();
+        return (valueAsJsonNode instanceof ArrayNode) && isValueAProductReference(valueAsJsonNode.elements().next());
+    }
+
+    private static boolean isValueAProductReference(@Nonnull final JsonNode valueAsJsonNode) {
+        if (valueAsJsonNode.isContainerNode()) {
+            final JsonNode typeIdNode = valueAsJsonNode.get("typeId");
```

Comment:
Put these strings as constant fields of this class.

Reply:
why? when it's only used here?

Reply:
vote up for @lojzatran comment. i checked - "typeId" is used 3 times in 2 classes

Reply:
@lojzatran @andrii-kovalenko-ct 1b7c07b